File: 1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

package main

// @lc code=start
func twoSum(nums []int, target int) []int {
	hashMap := make(map[int]int, 0)

	for i, num := range nums {
		rest, ok := hashMap[target-num]
		if ok {
			return []int{i, rest}
		}
		hashMap[num] = i
	}

	return make([]int, 0)
}

// @lc code=end

File: 1.两数之和.rs
/*
 * @lc app=leetcode.cn id=1 lang=rust
 *
 * [1] 两数之和
 */

use crate::structs::*;

// @lc code=start
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hash_map = std::collections::HashMap::<i32, i32>::new();

        for (i, num) in nums.into_iter().enumerate() {
            if let Some(rest_i) = hash_map.get(&(target - num)) {
                return vec![i as i32, *rest_i];
            }
            hash_map.insert(num, i as i32);
        }

        unreachable!("Only one answer")
    }
}
// @lc code=end
